### ISSUE TITLE
Image preview for has_many_attached broken since 0.3.x version

### ISSUE THREAD

**Issue by plabadille** [Dreamersoul/administrate-field-active_storage#80]
Hello,

It seems you broke image preview on `Field::ActiveStorage` for multiple files (`has_many_attached`) since 0.3.x versions.
This is because the `image?` method doesn't exists for `ActiveStorage::Attached::Many` :

```
Showing /bundle/ruby/2.5.0/gems/administrate-field-active_storage-0.3.7/app/views/fields/active_storage/_preview.html.erb where line #2 raised:

undefined method `image?' for #<ActiveStorage::Attached::Many:xxx>
```
Versions used :
- rails 6.0.4.4
- ruby 2.5.8
- administrate 0.16.0
- administrate-field-active_storage 0.3.7

I found two way of fixing it :

- not displaying preview on index : `Field::ActiveStorage.with_options(index_display_preview: false)`
- overriding your `active_storage/_index.html.erb` by mixing some parts of https://github.com/Dreamersoul/administrate-field-active_storage/blob/0.2.2/app/views/fields/active_storage/_index.html.erb, https://github.com/Dreamersoul/administrate-field-active_storage/blob/0.1.3/app/views/fields/active_storage/_index.html.erb and https://github.com/Dreamersoul/administrate-field-active_storage/blob/0.3.7/app/views/fields/active_storage/_index.html.erb :

`app/views/fields/active_storage/_index.html.erb`
```
<%#
# Image Index Partial
This partial renders an image attribute
to be displayed on a resource's index page.
By default, the attribute is rendered as an image tag.
## Local variables:
- `field`:
  An instance of [Administrate::Field::Image][1].
  A wrapper around the image url pulled from the database.
[1]: http://www.rubydoc.info/gems/administrate/Administrate/Field/Image
%>
<% if field.attached? %>
  <style> <%# figure out a way to remove this %>
  td img {
    max-height: unset !important;
  }
  </style>
  <%
    # from https://github.com/Dreamersoul/administrate-field-active_storage/blob/0.2.2/app/views/fields/active_storage/_index.html.erb
    # @since 0.3.x administrate-field-active_storage seems to have broke preview in index for `has_many_attached`
    attachments = Array(field.many? ? field.attachments : field.data)
  %>
  <% if field.index_display_preview? %>
    <% field.attachments.each do |attachment| %>
      <%= render partial: 'fields/active_storage/preview',
               locals: {
                   field: field,
                   attachment: attachment,
                   size: field.index_preview_size
               } %>
    <% end %>
  <% end %>
  <% if field.index_display_count? %>
    <div class="attachments-count">
      <%= pluralize(field.attachments.count,
                    I18n.t("administrate.fields.active_storage.attachment", default: 'Attachment')
          )
      %>
    </div>
  <% end %>
<% end %>
```

Did I miss something or it's a bug ?

**Comment by Dreamersoul** [comment]
It should work just fine, the field gets passed to the item partial correctly, did you restart the dev server after updating the gem? It could be that the field.rb file not loaded correctly after the update

**Comment by plabadille** [comment]
Yes I restarted multiple time. The error i'm seeing is your template _index got a field of type "ActiveStorage::Attached::Many" which doesn't have a method `image?`.
Your _index template doesn't  use the items partial, it use directly the preview one (and i think it may be the issue ?)

As requested a snippet of my dashboard file :
```
require "administrate/base_dashboard"

class RegistrationDocumentDashboard < Administrate::BaseDashboard
  # ATTRIBUTE_TYPES
  # a hash that describes the type of each of the model's fields.
  #
  # Each different type represents an Administrate::Field object,
  # which determines how the attribute is displayed
  # on pages throughout the dashboard.
  ATTRIBUTE_TYPES = {
    id: Field::Number,
    # please note that it's a special case : there's no column named as such
    # a method return the result
    name: Field::Text,
    # please note that it's a special case, there's no association
    # with this name, the result come from a method and using HasOne
    # field ensure us we display the proper resource name and so.
    user: Field::HasOne.with_options(
      class_name: 'Worker'
    ),
    files: Field::ActiveStorage,
    document_state: Field::Select.with_options(
      collection: RegistrationDocument.defined_document_states,
    ),
    expiry_date: Field::DateTime.with_options(
      # @see https://ruby-doc.org/stdlib-2.6.1/libdoc/date/rdoc/DateTime.html#method-i-strftime
      format: "%Y-%m-%d",
    ),
    created_at: Field::DateTime,
    updated_at: Field::DateTime,
  }.freeze

  # COLLECTION_ATTRIBUTES
  # an array of attributes that will be displayed on the model's index page.
  #
  # By default, it's limited to four items to reduce clutter on index pages.
  # Feel free to add, remove, or rearrange items.
  COLLECTION_ATTRIBUTES = [
    :id,
    :user,
    :name,
    :document_state,
    :files,
    :expiry_date,
    :created_at,
  ].freeze

  # SHOW_PAGE_ATTRIBUTES
  # an array of attributes that will be displayed on the model's show page.
  SHOW_PAGE_ATTRIBUTES = [
    :user,
    :name,
    :document_state,
    :expiry_date,
    :files,
    :created_at,
    :updated_at,
  ].freeze

  # FORM_ATTRIBUTES
  # an array of attributes that will be displayed
  # on the model's form (`new` and `edit`) pages.
  FORM_ATTRIBUTES = [
    :files,
    :document_state,
    :expiry_date,
  ].freeze

  # Overwrite this method to customize how documents are displayed
  # across all pages of the admin dashboard.

  def display_resource(document)
    "Document ##{document.id} - #{document.name}"
  end

  # @see https://github.com/Dreamersoul/administrate-field-active_storage#has_many_attached
  # permitted for has_many_attached
  def permitted_attributes
    super + [:files => []]
  end

  # disable user preloading because it's not a real association
  # but an instance method
  # @see https://github.com/thoughtbot/administrate/blob/v0.14.0/lib/administrate/base_dashboard.rb#L75
  def collection_includes
    attribute_includes(collection_attributes - [:user])
  end
end
```

**Comment by Dreamersoul** [comment]
As a workaround you can use the following option

```rb
  ATTRIBUTE_TYPES = {
    attachments: Field::ActiveStorage.with_options(index_display_preview: false),
    # ...
  }
```

But I still think this is an issue if you want you can push a PR to fix it

**Comment by plabadille** [comment]
Yes, when posting the issue I already found the two solutions : the one you juste told me (meaning i can't have preview in index anymore) or overriding the _index template which i prefer (the code i gave in my initial message was the fix, i just wanted to be sure it was a bug).

Ok so I will do the PR, thanks

**Comment by plabadille** [comment]
Thanks for merging, so this issue is fixed with the new 0.3.8 release :)
